internal/resources/metal/project/resource.go

```diff
@@ -3,311 +3,304 @@ package project
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
-	result = append(result, r)
+	// Retrieve the API client from the provider metadata
+	client := r.Meta.NewMetalClientForFramework(ctx, req.ProviderMeta)
 
-	return result
-}
+	// Extract the ID of the resource from the state
+	id := state.ID.ValueString()
+
+	// Handle BGP Config changes
+	bgpConfig, diags := handleBGPConfigChanges(ctx, client, &plan, &state, id)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-func resourceMetalProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*config.Config).NewMetalClientForSDK(d)
+	// Prepare Project update request based on the changes
 	updateRequest := metalv1.ProjectUpdateInput{}
-	if d.HasChange("name") {
-		pName := d.Get("name").(string)
-		updateRequest.Name = &pName
+	if state.Name != plan.Name {
+		updateRequest.Name = plan.Name.ValueStringPointer()
 	}
-	if d.HasChange("payment_method_id") {
-		pPayment := d.Get("payment_method_id").(string)
-		updateRequest.PaymentMethodId = &pPayment
+	if state.PaymentMethodID != plan.PaymentMethodID {
+		updateRequest.PaymentMethodId = plan.PaymentMethodID.ValueStringPointer()
 	}
-	if d.HasChange("backend_transfer") {
-		pBT := d.Get("backend_transfer").(bool)
-		updateRequest.BackendTransferEnabled = &pBT
+	if state.BackendTransfer != plan.BackendTransfer {
+		updateRequest.BackendTransferEnabled = plan.BackendTransfer.ValueBoolPointer()
 	}
-	if d.HasChange("bgp_config") {
-		o, n := d.GetChange("bgp_config")
-		oldarr := o.([]interface{})
-		newarr := n.([]interface{})
-		if len(newarr) == 1 {
-			bgpCreateRequest, err := expandBGPConfig(d)
-			if err != nil {
-				return diag.FromErr(err)
-			}
-
-			resp, err := client.BGPApi.RequestBgpConfig(ctx, d.Id()).BgpConfigRequestInput(*bgpCreateRequest).Execute()
-			if err != nil {
-				return diag.FromErr(equinix_errors.FriendlyErrorForMetalGo(err, resp))
-			}
-		} else {
-			if len(oldarr) == 1 {
-				m := oldarr[0].(map[string]interface{})
-
-				bgpConfStr := fmt.Sprintf(
-					"bgp_config {\n"+
-						"  deployment_type = \"%s\"\n"+
-						"  md5 = \"%s\"\n"+
-						"  asn = %d\n"+
-						"}", m["deployment_type"].(string), m["md5"].(string),
-					m["asn"].(int))
-
-				return diag.Errorf("BGP Config can not be removed from a project, please add back\n%s", bgpConfStr)
-			}
-		}
-	} else {
-		_, resp, err := client.ProjectsApi.UpdateProject(ctx, d.Id()).ProjectUpdateInput(updateRequest).Execute()
+
+	// NOTE (ocobles): adding UpdateProject inside this condition to replicate old behavior
+	// but it is not clear to me if it was a mistake. I think the project should be updated if
+	// has changes regardless of whether there are changes to the BGP configuration or not.
+	// Open discussion: https://github.com/equinix/terraform-provider-equinix/discussions/466
```

Comment:
I agree that this looks like a long-standing bug.  I say remove the check here so that it is possible to change BGP settings and core project settings in one plan/apply cycle, but we'll need to add checks here to make sure that the project settings _other than BGP_ are being changed so we don't send unnecessary update requests.